src/operator/nn/dropout-inl.h

```diff
@@ -262,7 +262,7 @@ class DropoutOp {
         Tensor<xpu, 1, unsigned>(reinterpret_cast<unsigned *>(workspace_ptr),
                                  Shape1(1), s);
       prnd->GetRandInt(random_number);
-      uint64_t seed_ = 17 + reinterpret_cast<uint64_t>(&random_number) % 4096;
+      uint64_t seed_ = 17 + reinterpret_cast<uint64_t>(random_number.dptr_) % 4096;
```

Comment:
random_number[0]

Reply:
i get `dropout-inl.h(265): error: invalid type conversion`

Reply:
dptr_ is a pointer and I don't think you mean to use its address here.

Reply:
I think it should be `uint64_t seed_ = 17 + static_cast<uint64_t>(random_number[0]) % 4096;`, because the type of `random_number[0]` is `unsigned`.

https://github.com/apache/incubator-mxnet/blob/master/3rdparty/mshadow/mshadow/tensor.h#L591